### PR TITLE
Sorting rewrite

### PR DESCRIPTION
**Pull request by laum18**
Re-did pivot table custom sorting. Now, the pivot table tree is sorted before row items are created.

### CODE REVIEW

File: EPPlus/Table/PivotTable/ExcelPivotTable.cs [hunk 1]
```diff
@@ -1294,10 +1382,11 @@ private void UpdateRowColumnItems(ExcelPivotTableRowColumnFieldCollection field,
 					//this.BuildRowItems(0, new List<Tuple<int, int>>(), pageFieldIndices, 0);
 
 					var root = this.BuildRowTree(pageFieldIndices);
+					this.SortTree(root);
 					this.BuildRowItems2(root, new List<Tuple<int, int>>());
 
 					clock.Stop();
-					System.IO.File.AppendAllText(@"C:\Users\ems\Downloads\time.txt", Environment.NewLine + clock.ElapsedMilliseconds.ToString());
+					//System.IO.File.AppendAllText(@"C:\Users\ems\Downloads\time.txt", Environment.NewLine + clock.ElapsedMilliseconds.ToString());
```
Comment (shawncrabtree) [inline EPPlus/Table/PivotTable/ExcelPivotTable.cs]: delete

File: EPPlus/Table/PivotTable/PivotItemTreeNode.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace OfficeOpenXml.Table.PivotTable
 	/// <summary>
 	/// Maintains state for a tree data structure that represents a pivot table.
 	/// </summary>
-	internal class PivotItemTreeNode
+	public class PivotItemTreeNode
```
Comment (shawncrabtree) [inline EPPlus/Table/PivotTable/PivotItemTreeNode.cs]: Why does this need to be public?

Comment (laum18) [reply EPPlus/Table/PivotTable/PivotItemTreeNode.cs]: It is being used in PivotTableCacheRecords to calculate the total used for sorting.

File: EPPlus/Table/PivotTable/ExcelPivotTable.cs (hunk 1 above)
Comment (laum18) [reply EPPlus/Table/PivotTable/ExcelPivotTable.cs]: Our next task is cleaning up building row items so this will be done next.

**Pull request merged**